### PR TITLE
[Lang] Add math module to support glsl-like functions

### PR DESCRIPTION
**Pull request by neozhaoliang**
Sorry for so many file changes, because I rebased my fork with the latest taichi-dev/master branch.

### CODE REVIEW

File: python/taichi/math/mathimpl.py [hunk 1]
```diff
@@ -0,0 +1,401 @@
+"""
+Math functions for glsl-like functions and other stuff.
+"""
+import taichi as ti
+
+pi = 3.1415926535897932
+e = 2.718281828459045
```
Comment (frostming) [inline python/taichi/math/mathimpl.py:7]: may be better to use the constants from `math` module.

Comment (neozhaoliang) [reply python/taichi/math/mathimpl.py:7]: Sure, will update this later.

**Pull request closed without merging**